align/align.go

```diff
@@ -197,15 +230,15 @@ func SmithWaterman(stringA string, stringB string, scoring Scoring) (int, string
 		}
 	}
 
-	return maxScore, alignA, alignB
+	return maxScore, alignA, alignB, nil
 }
 
-func score(a, b byte, scoring Scoring) int {
-	if a == b {
-		return scoring.Match
-	} else {
-		return scoring.Mismatch
+func score(a, b byte, scoring Scoring) (int, error) {
```

Comment:
Maybe score should be a function of the Scoring struct.

Reply:
Good point!